app/src/main/java/com/elavon/converge/util/RequestUtil.java

```diff
@@ -0,0 +1,29 @@
+package com.elavon.converge.util;
+
+public class RequestUtil {
+
+    private static final String HYPHEN = "-";
+    private static final String DEFAULT_POYNT_USER = "PoyntUser";
+
+    public static String getDeviceUserValue(String firstName, String lastName, String nickname) {
+        StringBuilder userBuilder = new StringBuilder();
+        if (nickname != null && nickname.length() > 0) {
+            if((firstName != null && firstName.length() > 0) ||
+                    (lastName != null && lastName.length() > 0)) {
+                userBuilder.append(nickname).append(HYPHEN);
+            }
+        }
+        if (firstName != null && firstName.length() > 0) {
+            if(lastName != null && lastName.length() > 0) {
+                userBuilder.append(firstName).append(HYPHEN);
```

Comment:
rather than appending - prepend so when there is no last name the user won't be "nickname-firstname-"